src/sagemaker/inference_recommender/inference_recommender_mixin.py

```diff
@@ -149,12 +148,26 @@ def right_size(
 
         self._init_sagemaker_session_if_does_not_exist()
 
+        model_name = None
+        if isinstance(self, sagemaker.model.FrameworkModel):
+
+            unique_tail = uuid.uuid4()
+            model_name = "SMPYTHONSDK-" + str(unique_tail)
+
+            self.sagemaker_session.create_model(
```

Comment:
should we log the creation of model to the customer?

Reply:
do we know if we are missing vpcConfig, networkIsolation and other options?

the `deploy()` method seems to use this which is a private member of Model. Could we use that?

https://github.com/aws/sagemaker-python-sdk/blob/5ca7f281c97693c2fcd10eef3b8ffbc75b76e0a9/src/sagemaker/model.py#L638

Reply:
leverage this on deletion as well: https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/model.py#L1320

Reply:
lets also be very clear on what the side effects (if any) are of calling `_create_sagemaker_model` and in turn setting `base_name` and `name` for a model that will be deleted

Reply:
the `def _create_sagemaker_model( )` will create a model with the `self.name` variable, but I think we should allocate and release our own resources to be used for IR job and don't use customer's resources under the hood. If customer already created a model then we don't want to delete it after they call `right_size()`
We can still do:
```
create_model_args = dict(
                name=temp_model_name,
                role=self.role,
                container_defs=None,
                primary_container=self.prepare_container_def(),
                vpc_config=self.vpc_config,
                enable_network_isolation=self.enable_network_isolation()
            )

            self.sagemaker_session.create_model(**create_model_args)
```
Similar to:
https://github.com/aws/sagemaker-python-sdk/blob/5ca7f281c97693c2fcd10eef3b8ffbc75b76e0a9/src/sagemaker/model.py#L688

Reply:
Same applies for delete model, sicne we are using our `temp_model_name` we can't call the delete method in leverage this on deletion as well: https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/model.py#L1320 because it deletes the model with `self.name`
 which is customer allocated/managed resource that we should avoid modifying